lms/services/grouping/_plugin/blackboard.py

```diff
@@ -0,0 +1,57 @@
+from lms.models import Grouping
+from lms.services import ExternalRequestError
+from lms.services.grouping.service import GroupingServicePlugin
+from lms.views.api.exceptions import GroupError
+
+
+class BlackboardStudentNotInGroup(GroupError):
+    """Student doesn't belong to any of the groups in a group set assignment."""
+
+    error_code = "blackboard_student_not_in_group"
+
+
+class BlackboardGroupSetEmpty(GroupError):
+    """Canvas GroupSet doesn't contain any groups."""
+
+    error_code = "blackboard_group_set_empty"
+
+
+class BlackboardGroupSetNotFound(GroupError):
+    error_code = "blackboard_group_set_not_found"
```

Comment:
Looks like we could sort this out with something like:

`raise GroupError("blackboard_group_set_not_found")`

?